### PR TITLE
feat: separate CollectorNameForm from CreateCollectorStep2 & add CollectorNameEditModal for CollectorDetailPage

### PR DESCRIPTION
**Pull request by WANZARGEN**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [x] New feature
- [ ] Bug fixes
- [ ] Feature improvement
- [x] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [ ] mirinae
  - [ ] etc 

- [ ] Apps
  - [ ] storybook
  - [x] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Description

Please review in order by commits.

https://github.com/cloudforet-io/console/assets/26986739/90225403-b14b-4fd2-9b92-ef380cdb55bc

### CODE REVIEW

File: apps/web/src/services/asset-inventory/collector/collector-create/modules/CreateCollectorStep2.vue [hunk 1]
```diff
@@ -2,19 +2,7 @@
...
-            >
-                <template #default="{invalid}">
-                    <p-text-input :value="name"
-                                  class="block"
-                                  :invalid="invalid"
-                                  @update:value="setForm('name', $event)"
-                    />
-                </template>
-            </p-field-group>
+            <collector-name-form @update:isValid="handleUpdateIsValid" />
```
Comment (yuda110) [inline apps/web/src/services/asset-inventory/collector/collector-create/modules/CreateCollectorStep2.vue:5]: praise: I like this!

**Review by yuda110 — APPROVED** [review]
LGTM!! very readable

File: apps/web/src/services/asset-inventory/collector/collector-create/modules/CreateCollectorStep2.vue [hunk 2]
```diff
@@ -88,49 +75,31 @@ const state = reactive({
...
-
-
 /* event */
 
 const handleClickPrevButton = () => {
     state.deleteModalVisible = true;
 };
 const handleClickNextButton = () => {
-    collectorFormStore.setName(name.value);
+    collectorFormStore.setName(collectorFormState.name);
```
Comment (sulmoJ) [inline apps/web/src/services/asset-inventory/collector/collector-create/modules/CreateCollectorStep2.vue:92]: opinion: I think you can remove this code!

**Review by sulmoJ — APPROVED** [review]
I left a comment!
LGTM!

**Pull request merged**